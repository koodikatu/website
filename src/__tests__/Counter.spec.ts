import '@testing-library/jest-dom';
import { render, fireEvent } from '@testing-library/svelte';
import Counter from '../lib/Counter.svelte';

it('renders correctly', () => {
	const { container } = render(Counter);
	expect(container).toMatchSnapshot();
});

it('it works', async () => {
	const { getByTestId } = render(Counter);

	const increment = getByTestId('increment');
	const decrement = getByTestId('decrement');
	const counter = getByTestId('counter-value');

	expect(counter).toHaveTextContent('0');

	await fireEvent.click(increment);
	await fireEvent.click(increment);
	await fireEvent.click(increment);
	await fireEvent.click(decrement);

	expect(counter).toHaveTextContent('2');
});
