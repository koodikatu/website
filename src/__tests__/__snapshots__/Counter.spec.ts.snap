// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<body>
  <div>
    <div
      class="counter svelte-9qsz00"
    >
      <button
        aria-label="Decrease the counter by one"
        class="svelte-9qsz00"
        data-testid="decrement"
      >
        <svg
          aria-hidden="true"
          class="svelte-9qsz00"
          viewBox="0 0 1 1"
        >
          <path
            class="svelte-9qsz00"
            d="M0,0.5 L1,0.5"
          />
        </svg>
      </button>
       
      <div
        class="counter-viewport svelte-9qsz00"
      >
        <div
          class="counter-digits svelte-9qsz00"
          style="transform: translate(0, 0%);"
        >
          <strong
            aria-hidden="true"
            class="svelte-9qsz00"
            style="top: -100%;"
          >
            1
          </strong>
           
          <strong
            class="svelte-9qsz00"
            data-testid="counter-value"
          >
            0
          </strong>
        </div>
      </div>
       
      <button
        aria-label="Increase the counter by one"
        class="svelte-9qsz00"
        data-testid="increment"
      >
        <svg
          aria-hidden="true"
          class="svelte-9qsz00"
          viewBox="0 0 1 1"
        >
          <path
            class="svelte-9qsz00"
            d="M0,0.5 L1,0.5 M0.5,0 L0.5,1"
          />
        </svg>
      </button>
    </div>
  </div>
</body>
`;
